core/src/main/java/io/fabric8/maven/core/util/ProfileUtil.java

```diff
@@ -34,105 +35,148 @@
-                    ret.put(profile.getName(), profile);
-                }
-            }
-        }
-        return ret;
-    }
 
     /**
-     * Lookup profiles from a given directory
+     * Lookup profiles from a given directory and merge it with a profile of the
+     * same name found in the classpath
      *
      * @param name name of the profile to lookup
      * @param directory directory to lookup
      * @return Profile found or null
      * @throws IOException if somethings fails during lookup
      */
     public static Profile lookup(String name, File directory) throws IOException {
+        // First check from the classpath, these profiles are used as a basis
+        Profile cpProfile = readProfileFromClasspath(name);
+
         File profileFile = findProfileYaml(directory);
         if (profileFile != null) {
             List<Profile> profiles = fromYaml(new FileInputStream(profileFile));
             for (Profile profile : profiles) {
                 if (profile.getName().equals(name)) {
-                    return profile;
+                    return mergeProfiles(cpProfile, profile);
+                }
+            }
+        }
+        return cpProfile;
+    }
+
+    private static ProcessorConfig extractProcesssorConfiguration(ProcessorConfigurationExtractor extractor,
+                                                                 String profile,
+                                                                 File resourceDir) throws IOException {
+        Profile profileFound = findProfile(profile, resourceDir);
+        return extractor.extract(profileFound);
+    }
+
+
+    private static Profile mergeProfiles(Profile ... profiles) {
+        Profile ret = null;
+        for (Profile profile : profiles) {
+            if (profile != null) {
+                if (ret == null) {
+                    ret = new Profile(profile);
+                } else {
+                    ret = new Profile(ret, profile);
+                }
+            }
+        }
+        return ret;
+    }
+
+    // Read all default profiles first, then merge in custom profiles found on the classpath
+    private static Profile readProfileFromClasspath(String name) throws IOException {
+        Profile basePofile = mergeProfiles(readAllFromClasspath(name, "default"));
+        Profile addOnProfile = mergeProfiles(readAllFromClasspath(name, ""));
+        return mergeProfiles(basePofile, addOnProfile);
+    }
+
+    /**
+     * Read all profiles found in the classpath.
+     *
+     * @param name name of the profile to lookup
+     * @param ext to use (e.g. 'default' for checking 'profile-default.yml'. Can also be null or empty.
+     * @return all profiles with this name stored in files with this extension
+     *
+     * @throws IOException if reading of a profile fails
+     */
+    public static Profile[] readAllFromClasspath(String name, String ext) throws IOException {
+        List<Profile > ret = new ArrayList<>();
```

Comment:
Ah, just saw that I made my old error again: If a comparator returns 0, then both elements must be also equal. If not, they override each other in the TreeSet. In our case the order can be the same for both processors (0 by default), so I need an extra check whether its the same.